src/components/text/text.styles.ts

```diff
@@ -0,0 +1,339 @@
+    headingFontSize: ReturnType<typeof euiTitle>['fontSize']
+  ) => {
+    let marginBottom = parseFloat(String(fontSize)) / 2;
+    if (measurement === 'em') {
+      marginBottom = marginBottom / parseFloat(String(headingFontSize));
+    }
+    return `${marginBottom}${measurement}`;
+  };
+
+  return `
+    font-size: ${fontSize};
+    line-height: ${lineHeight};
+
+    h1 {
+      font-size: ${headings.h1.fontSize};
+      line-height: ${headings.h1.lineHeight};
+    }
+    h1:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h1.fontSize))}
+    }
+
+    h2 {
+      font-size: ${headings.h2.fontSize};
+      line-height: ${headings.h2.lineHeight};
+    }
+    h2:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h2.fontSize))}
+    }
+    * + h2 {
+      ${logicalCSS('margin-top', headingMarginTop(headings.h2.fontSize))}
+    }
+
+    h3 {
+      font-size: ${headings.h3.fontSize};
+      line-height: ${headings.h3.lineHeight};
+    }
+    h3:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h3.fontSize))}
+    }
+    * + h3 {
+      ${logicalCSS('margin-top', headingMarginTop(headings.h3.fontSize))}
+    }
+
+    h4 {
+      font-size: ${headings.h4.fontSize};
+      line-height: ${headings.h4.lineHeight};
+    }
+    h4:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h4.fontSize))}
+    }
+    * + h4 {
+      ${logicalCSS('margin-top', headingMarginTop(headings.h4.fontSize))}
+    }
+
+    h5 {
+      font-size: ${headings.h5.fontSize};
+      line-height: ${headings.h5.lineHeight};
+    }
+    h5:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h5.fontSize))}
+    }
+    * + h5 {
+      ${logicalCSS('margin-top', headingMarginTop(headings.h5.fontSize))}
+    }
+
+    h6 {
+      font-size: ${headings.h6.fontSize};
+      line-height: ${headings.h6.lineHeight};
+    }
+    h6:not(:last-child) {
+      ${logicalCSS('margin-bottom', headingMarginBottom(headings.h6.fontSize))}
+    }
+    * + h6 {
+      ${logicalCSS('margin-top', headingMarginTop(headings.h6.fontSize))}
+    }
+
+    p,
+    ul,
+    ol,
+    dl,
+    blockquote,
+    img,
+    pre {
+      ${logicalCSS('margin-bottom', lineHeightSize)}
+    }
+
+    ul,
+    ol {
+      ${logicalCSS('margin-left', lineHeightSize)}
+    }
+  
+    blockquote {
+      font-size: ${fontSize};
+      padding: ${lineHeightSize};
+    }
+
+    dd + dt {
+      ${logicalCSS('margin-top', fontSize)}
+    }
+
+    dt,
+    .eui-definitionListReverse dd {
+      font-size: ${fontSize};
+      line-height: ${lineHeight};
+    }
+
+    .eui-definitionListReverse dt {
+      font-size: ${euiFontSize('xs', euiTheme, options).fontSize};
+      color: ${euiTheme.colors.text};
+    }
+
+    small {
+      font-size: ${euiFontSize('s', euiTheme, options).fontSize};
+    }
+
+    pre {
+      padding: ${fontSize};
+    }
+
+    code {
```

Comment:
Still need this to exclude our own EuiCode/EuiCodeBlock elements

```suggestion
    code:not(.euiCode):not(.euiCodeBlock__code)  {
```